src/server/scripts/World/npcs_special.cpp

```diff
@@ -580,6 +580,107 @@ class npc_torch_tossing_target_bunny_controller : public CreatureScript
 };
 
 /*######
+## npc_midsummer_bunny_pole
+######*/
+
+enum RibbonPoleData
+{
+    GO_RIBBON_POLE              = 181605,
+    SPELL_RIBBON_DANCE_COSMETIC = 29726,
+    SPELL_RED_FIRE_RING         = 46836,
+    SPELL_BLUE_FIRE_RING        = 46842,
+    EVENT_CAST_RED_FIRE_RING    = 1,
+    EVENT_CAST_BLUE_FIRE_RING   = 2
+};
+
+class npc_midsummer_bunny_pole : public CreatureScript
+{
+public:
+    npc_midsummer_bunny_pole() : CreatureScript("npc_midsummer_bunny_pole") { }
+
+    struct npc_midsummer_bunny_poleAI : public ScriptedAI
+    {
+        npc_midsummer_bunny_poleAI(Creature* creature) : ScriptedAI(creature) { Reset(); }
```

Comment:
Usually there's a Initialize() function that sets the fields to default values called from the ctor and from Reset(), so you don't have to duplicate code.
Please make sure all class fields are initialized (i.e. "running")